### PR TITLE
Add galaxy to user agent

### LINKED ISSUE #17995: Specific URL fails to be downloaded from different galaxy instances
**Describe the bug**

The following GTF URL from 10x fails to be downloaded from 3 different galaxy instances (Org, EU, private 22.05 and another private one on a different version). The download works from browsers or wget in the same servers. The file is fairly big (~9 GBs), but the error comes in second, so it doesn't seem to be size related. Or no error message mentions that. I suspect that something in the client definition provided in the HTTP requests by Galaxy might be upsetting the 10x hosting somehow? Or in way at least that wget doesn't upset the same host.

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: multiple above 21.x
Commit: 

**Browser and Operating System**
Operating System: macOS and Windows
Browser: Chrome

**To Reproduce**
1. Go to 'upload data'
2. Paste the URL https://cf.10xgenomics.com/supp/cell-exp/refdata-gex-GRCm39-2024-A.tar.gz in the Paste/fetch data part

**Expected behavior**
The file should download correctly, however different errors are shown (HTTP 403, 'fetch_url_allowlist'

**Screenshots**

image
image

**Additional context**

### PR DESCRIPTION
**Pull request by mvdbeek**
Builds on #17578 but avoids global manipulation of requests. Should eventually also fix https://github.com/galaxyproject/galaxy/issues/17995

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by jmchilton** [comment]
Very cool typing stuff in there! I'll have to revisit that and dig in more - looks great.

File: lib/galaxy/tools/data_fetch.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
     stream_to_file,
     stream_url_to_file,
 )
+from galaxy.util import user_agent  # noqa: F401
```
Comment (nsoranzo) [inline lib/galaxy/tools/data_fetch.py:29]: This is not needed any more as well, I think?

File: lib/galaxy/app.py [hunk 2]
```diff
@@ -133,6 +133,7 @@
     build_tours_registry,
     ToursRegistry,
 )
+from galaxy.util import user_agent  # noqa: F401
```
Comment (nsoranzo) [inline lib/galaxy/app.py:136]: This is not needed any more, right?

File: lib/galaxy/tools/error_reports/plugins/gitlab.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 import os
 import urllib.parse
 
-import requests
+from galaxy.util import requests
```
Comment (nsoranzo) [inline lib/galaxy/tools/error_reports/plugins/gitlab.py:7]: Should be moved after the gitlab import try/except.

File: scripts/api/fetch_to_library.py [hunk 4]
```diff
@@ -1,9 +1,10 @@
 import argparse
 import json
 
-import requests
 import yaml
 
+from galaxy.util import requests
```
Comment (nsoranzo) [inline scripts/api/fetch_to_library.py:6]: This doesn't make much sense for this and the other scripts in `scripts/api/` as they are meant to be example for how to connect to a Galaxy instance (not from a Galaxy server).

File: scripts/tool_shed/api/common.py [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import requests
+from galaxy.util import requests
```
Comment (nsoranzo) [inline scripts/tool_shed/api/common.py:1]: This one was also not needed, I think.

File: lib/galaxy/util/__init__.py [hunk 6]
```diff
@@ -68,6 +67,8 @@
     Self,
 )
 
+from galaxy.util import requests
```
Comment (nsoranzo) [inline lib/galaxy/util/__init__.py:70]: Should be moved after the try/except imports (and can be rewritten as `from . import requests`).

**Review by nsoranzo — COMMENTED** [review]
Sorry for belated review, if you agree these comments make sense, I can open a PR to fix them.

**Pull request merged**